### PR TITLE
1.6/1301 edit workers

### PR DESCRIPTION
**Pull request by alexalala**
Just needs extra 'neighbours' logic and PUT integrated

https://trello.com/c/pvXxhJTw/1301-edit-workers

🍫

### CODE REVIEW

File: app/Http/Controllers/Service/Admin/CentreUsersController.php [hunk 1]
```diff
@@ -36,6 +37,54 @@ public function create()
...
+                    }
+                });
+            });
+
+        return view('service.centreusers.edit', compact('worker', 'centresBySponsor'));
+    }
+    
+    public function update($id)
+    {
+        //TBFI
```
Comment (katjam) [inline app/Http/Controllers/Service/Admin/CentreUsersController.php]: We know this is not ready?

File: resources/views/service/centreusers/edit.blade.php [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+<div id="container">
+    @include('service.includes.sidebar')
+    <div id="main-content">
+
+        <h1>Edit a Children's Centre Worker</h1>
+
+        <p>Use the form below to amend a Children's Centre Worker's details.</p>
+
+        <!-- ADD NEW POST ROUTE HERE eg. method="POST" action="{{ route('admin.vouchers.storebatch') }}" -->
```
Comment (katjam) [inline resources/views/service/centreusers/edit.blade.php]: Not ready yet?

File: resources/views/service/centreusers/edit.blade.php [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+                </div>
+                <div class="select">
+                    <label for="worker_centre">Home Centre</label>
+                    <select name="worker_centre" id="worker_centre" class="{{ $errors->has('worker_centre') ? 'error' : '' }}" required>
+                        @foreach($centresBySponsor as $sponsor)
+                        <optgroup label="{{ $sponsor->name }}">
+                            @foreach ($sponsor->centres as $centre)
+                                <option value="{{ $centre->id }}"
+                                        @if($centre->selected === "home")
+                                        SELECTED
```
Comment (katjam) [inline resources/views/service/centreusers/edit.blade.php]: Do we need shouty caps?

File: resources/views/service/centreusers/edit.blade.php [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+                if (data.length > 0) {
+                    boxes = $.map(data, function(obj) {
+                        var div = $('<div class="checkbox-group">');
+
+                        // Make an input
+                        $('<input>')
+                            .attr({
+                                type    : 'checkbox',
+                                name    : 'alternate_centres[]',
+                                value   : obj.id,
```
Comment (katjam) [inline resources/views/service/centreusers/edit.blade.php]: spaces here before o and next line after + but totally not important.

File: resources/views/service/centreusers/edit.blade.php [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+                        <p><strong>Set Neighbours as Alternatives</strong></p>
+                        <div id="centres" class="checkboxes">
+                        </div>
+                    </div>
+                </div>
+            </div>
+            <button type="submit" id="createWorker">Add worker</button>
+        </form>
+        <script>
+            function buildCheckboxes() {
```
Comment (katjam) [inline resources/views/service/centreusers/edit.blade.php]: Would it be better to move to using vue templates for the js related pages? I've not tried - not sure how the laravel integration works but the theory I though was to replace blade stuff?

File: resources/views/service/centreusers/edit.blade.php [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+                    //if it's not checked, remove attribute
+                    : centre.removeAttr('data-workercentre')
+                ;
+                // rebuild the checkboxes.
+                buildCheckboxes();
+            }
+
+            // Start the page
+            $(document).ready(
+                function () {
```
Comment (katjam) [inline resources/views/service/centreusers/edit.blade.php]: So sad about all the `jquery` but :man_shrugging:

File: app/Http/Controllers/Service/Admin/CentreUsersController.php (hunk 1 above)
Comment (charlesstrange2) [reply app/Http/Controllers/Service/Admin/CentreUsersController.php]: yes. am implementing in another branch.

File: resources/views/service/centreusers/edit.blade.php (hunk 2 above)
Comment (charlesstrange2) [reply resources/views/service/centreusers/edit.blade.php]: again, in the submit/update handler branch.

File: resources/views/service/centreusers/edit.blade.php (hunk 3 above)
Comment (charlesstrange2) [reply resources/views/service/centreusers/edit.blade.php]: I though it was the standard for fixed attributes? maybe not... will check.

File: resources/views/service/centreusers/edit.blade.php (hunk 5 above)
Comment (charlesstrange2) [reply resources/views/service/centreusers/edit.blade.php]: The overhaead for the amount of script would be _insane_.

This is not a project where an SPA is required.

File: resources/views/service/centreusers/edit.blade.php (hunk 6 above)
Comment (charlesstrange2) [reply resources/views/service/centreusers/edit.blade.php]: "just enough javascript"

File: app/Http/Controllers/Service/Admin/CentreUsersController.php [hunk 7]
```diff
@@ -36,6 +37,54 @@ public function create()
...
+     * @param $id
+     * @return Factory|View
+     */
+    public function edit($id)
+    {
+        // Find the worker or throw a 500
+        $worker = CentreUser::findOrFail($id);
+
+        // Get the homeCentreId
+        $homeCentreId = $worker->homeCentre->first()->id;
```
[the remark this answers is not in the document]
Comment (charlesstrange2) [reply app/Http/Controllers/Service/Admin/CentreUsersController.php]: in theory, there should never be a situation where we don't have a homecentre.

File: resources/views/service/centreusers/edit.blade.php (hunk 3 above)
Comment (charlesstrange2) [reply resources/views/service/centreusers/edit.blade.php]: huh, apparently not, ok, will change.

**Pull request merged**